deployment/network-operator/templates/upgrade-crd-hook.yaml

```diff
@@ -84,12 +84,12 @@ spec:
             - /bin/sh
             - -c
             - >
-              kubectl apply -f /crds/network-operator;
+              /apply-crds --crds-dir=/crds/network-operator;
```

Comment:
nit: maybe we should adjust the script to take a list of dirs instead and split this in `command` and `args`

Reply:
I'll take a look at this, good idea 👍🏻

Reply:
Implemented and tested locally in a kind cluster

```
+ network-operator-upgrade-crd-r5hmb › upgrade-crd
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Apply CRDs from file: /crds/network-operator/k8s.cni.cncf.io_networkattachmentdefinitions_crd.yaml
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Update CRD network-attachment-definitions.k8s.cni.cncf.io
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Apply CRDs from file: /crds/network-operator/mellanox.com_hostdevicenetworks.yaml
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Update CRD hostdevicenetworks.mellanox.com
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Apply CRDs from file: /crds/network-operator/mellanox.com_ipoibnetworks.yaml
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Update CRD ipoibnetworks.mellanox.com
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Apply CRDs from file: /crds/network-operator/mellanox.com_macvlannetworks.yaml
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Update CRD macvlannetworks.mellanox.com
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Apply CRDs from file: /crds/network-operator/mellanox.com_nicclusterpolicies.yaml
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Update CRD nicclusterpolicies.mellanox.com
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Apply CRDs from file: /crds/node-feature-discovery/nfd-api-crds.yaml
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Update CRD nodefeatures.nfd.k8s-sigs.io
network-operator-upgrade-crd-r5hmb upgrade-crd 2024/10/31 09:29:32 Update CRD nodefeaturerules.nfd.k8s-sigs.io
- network-operator-upgrade-crd-r5hmb › upgrade-crd
```

```
    Command:
      /apply-crds
    Args:
      --crds-dir=/crds/network-operator
      --crds-dir=/crds/node-feature-discovery
```